contrib/verify-binaries/verify.py

```diff
@@ -512,9 +532,9 @@ def cleanup():
         return sigs_status
 
     # Extract hashes and filenames
-    hashes_to_verify = parse_sums_file(SUMS_FILENAME, [os_filter])
+    hashes_to_verify = parse_sums_file(SUMS_FILENAME, [version_arch, os_filter])
```

Comment:
As mentioned earlier, some values of `version_arch` and `os_filter` are still going to download multiple files but should not.
Such as:
"27.0-arm" fetches arm64 as well
"27.0-arm64" fetches both .zip and .tar.gz
"27.0-x86_64-apple-darwin" fetches both .zip and .tar.gz
"27.0-win64" fetches both setup.exe and .zip
etc.

So we need to be able to specify more than this.

That's why I suggested a simpler approach that doesn't try to understand what it's matching and just gets the matching or partially matching file name(s) in SHA256SUMS I feel that gives users more flexibility and is less code to maintain.